extensions/guides/third-party-libraries.md

```diff
@@ -0,0 +1,211 @@
+
+Once the library finishes loading, the global function is replaced with a new one that bypasses the queue and instead immediately processes any future calls to the function.
+
+### Base code example
+
+The following JavaScript is an example of an unminified base code for the [Pinterest conversion tag](https://developers.pinterest.com/docs/ad-tools/conversion-tag/?), which will be referenced later in this document to demonstrate how it can be adapted for different implementation strategies in Platform Launch:
+
+```js
+!function(scriptUrl) {
+  if (!window.pintrk) {
+    window.pintrk = function() {
+      window.pintrk.queue.push(
+        Array.prototype.slice.call(arguments)
+      );
+    };
+    window.pintrk.queue = []; 
+    window.pintrk.version = '3.0';
+    var scriptElement = document.createElement('script');
+    scriptElement.async = true;
+    scriptElement.src = scriptUrl;
+    var firstScriptElement = 
+      document.getElementsByTagName('script')[0];
+    firstScriptElement.parentNode.insertBefore(
+      scriptElement, firstScriptElement
+    );
+  }
+}('https://s.pinimg.com/ct/core.js');
+pintrk('load', 'YOUR_TAG_ID');
+pintrk('page');
+```
+
+In summary, the base code above provides an [immediately invoked function expression (IIFE)](https://developer.mozilla.org/en-US/docs/Glossary/IIFE) that creates a global function to interact with the library (`window.pintrk`). It also assigns a `scriptURL` variable the value of `https://s.pinimg.com/ct/core.js`, which is where the library is located. As explained earlier, any functions that are called before the library has loaded are pushed into a queue (`window.pintrk.queue`) to be executed in sequence once the library is available.
+
+The following part of the base code is the most relevant when it comes to understanding how the library loads on your site:
+
+```js
+var scriptElement = document.createElement("script");
+scriptElement.async = true;
+scriptElement.src = scriptUrl;
+var firstScriptElement = 
+  document.getElementsByTagName("script")[0];
+firstScriptElement.parentNode.insertBefore(
+  scriptElement, firstScriptElement
+);
+```
+
+The base code creates a script element, sets it to load asynchronously, and sets the `src` URL to `https://s.pinimg.com/ct/core.js`. It then adds the script element to the document by inserting it before the first script element already in the document.
+
+## Implementation options in Platform Launch
+
+The sections below demonstrate the different ways you can load vendor libraries in your extensions, using the Pinterest base code shown previously as an example. Each of these examples involve creating an [action type](/extensions/reference/action-types/) that loads the library on your website.
```

Comment:
Maybe a quick note here to say that you don't have to do it in an action type, that's just what we've chosen to do for these examples?